collector/process.go

```diff
@@ -177,13 +177,24 @@ type Win32_PerfRawData_PerfProc_Process struct {
 	WorkingSetPrivate       uint64
 }
 
+type WorkerProcess struct {
+	AppPoolName string
+	ProcessId   uint32
+}
+
 func (c *ProcessCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
 	var dst []Win32_PerfRawData_PerfProc_Process
 	q := queryAllWhere(&dst, c.queryWhereClause)
 	if err := wmi.Query(q, &dst); err != nil {
 		return nil, err
 	}
 
+	var dst_wp []WorkerProcess
+	q_wp := queryAll(&dst_wp)
+	if err := wmi.QueryNamespace(q_wp, &dst_wp, "root\\WebAdministration"); err != nil {
+		dst_wp = nil
```

Comment:
`dst_wp` is already nil, so you could skip this check entirely

Reply:
good point.  Removed the check.